### PR TITLE
Incremental backup with none-persistent bitmap

### PR DESCRIPTION
**Pull request by zhencliu**
Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1861955

### CODE REVIEW

**Comment by zhencliu** [comment]
(1/1) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_non_persistent_bitmap.local_target: PASS (85.71 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-08-05T23.50-189a8f7/results.html
JOB TIME   : 90.98 s

**Comment by zhencliu** [comment]
for iscsi and nbd backend, sometimes, case failed, i.e. count is greater than 0 even when no writing on data disk.

**Comment by zhencliu** [comment]
hi @YongxueHong @aliang123 can you please help to review the test case, thanks much.

**Comment by zhencliu** [comment]
nbd:
 (1/1) Host_RHEL.m8.u3.product_av.nbd.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_non_persistent_bitmap.local_target: PASS (233.56 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-08-06T05.09-904b7c9/results.html
JOB TIME   : 239.12 s

File: qemu/tests/cfg/blockdev_inc_backup_non_persistent_bitmap.cfg [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    image_format_data = qcow2
+    image_format_base = qcow2
+
+    image_name_data = data
+    image_name_base = base
+    rebase_mode = unsafe
+    check_bitmaps = 'bitmaps:'
+
+    variants:
```
Comment (aliang123) [inline qemu/tests/cfg/blockdev_inc_backup_non_persistent_bitmap.cfg:27]: After our discussion last week, please help to update the cfg here.

Comment (zhencliu) [reply qemu/tests/cfg/blockdev_inc_backup_non_persistent_bitmap.cfg]: done

**Comment by zhencliu** [comment]
(1/1) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_non_persistent_bitmap: PASS (81.92 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-08-10T04.05-6ecad92/results.html
JOB TIME   : 87.66 s

 (1/1) Host_RHEL.m8.u3.product_av.iscsi_direct.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_non_persistent_bitmap: PASS (90.04 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-08-10T04.07-8f922f8/results.html
JOB TIME   : 95.29 s

**Comment by aliang123** [comment]
cmd: 
   python3 ConfigTest.py --guestname=RHEL.8.3.0 --platform=x86_64 --testcase=blockdev_inc_backup_non_persistent_bitmap --machines=q35
result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_non_persistent_bitmap.q35: PASS (52.95 s)
   
Acked-by: Aihua Liang <aliang@redhat.com>

**Comment by zhencliu** [comment]
hi @ngu-niny , can you please help to verify this patch on ppc, thanks.

**Comment by ngu-niny** [comment]
Test PASS on ppc:

(1/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_non_persistent_bitmap: PASS (121.52 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**